openslides/utils/autoupdate.py

```diff
@@ -461,41 +120,48 @@ def __call__(self, request: Any) -> Any:
         response = self.get_response(request)
 
         bundle = autoupdate_bundle.pop(thread_id)  # type: Dict[str, CollectionElement]
-        # If currently there is an open database transaction, then the
-        # send_autoupdate function is only called, when the transaction is
-        # commited. If there is currently no transaction, then the function
-        # is called immediately.
-        transaction.on_commit(lambda: send_autoupdate(bundle.values()))
+        async_to_sync(send_autoupdate)(bundle.values())
         return response
 
 
-def send_autoupdate(collection_elements: Iterable[CollectionElement]) -> None:
+async def send_autoupdate(collection_elements: Iterable[CollectionElement]) -> None:
     """
     Helper function, that sends collection_elements through a channel to the
     autoupdate system.
 
+    Also updates the redis cache.
+
     Does nothing if collection_elements is empty.
     """
     if collection_elements:
-        send_or_wait(
-            Channel('autoupdate.send_data_projector').send,
-            to_channel_message(collection_elements))
-        send_or_wait(
-            Channel('autoupdate.send_data_site').send,
-            to_channel_message(collection_elements))
-
-
-def get_startup_collections() -> Generator[Collection, None, None]:
-    """
-    Returns all Collections that should be send to the user at startup
-    """
-    for app in apps.get_app_configs():
-        try:
-            # Get the method get_startup_elements() from an app.
-            # This method has to return an iterable of Collection objects.
-            get_startup_elements = app.get_startup_elements
-        except AttributeError:
-            # Skip apps that do not implement get_startup_elements.
-            continue
+        cache_elements = {}  # type: Dict[str, Optional[Dict[str, Any]]]
+        for element in collection_elements:
+            element_id = get_element_id(element.collection_string, element.id)
+            if element.is_deleted():
+                cache_elements[element_id] = None
+            else:
+                cache_elements[element_id] = element.get_full_data()
 
-        yield from get_startup_elements()
+        if not getattr(settings, 'SKIP_CACHE', False):
+            # Hack for django 2.0 and channels 2.1 to stay in the same thread.
+            # This is needed for the tests.
+            change_id = await element_cache.change_elements(cache_elements)
+        else:
+            change_id = 1
+
+        channel_layer = get_channel_layer()
+        # TODO: don't await. They can be send in parallel
```

Comment:
Maybe you can use https://docs.python.org/3/library/asyncio-task.html#asyncio.gather here. See https://docs.python.org/3/library/asyncio-task.html#example-parallel-execution-of-tasks

Reply:
I will add this performance improvement in another pull request. I am not sure if this can be used here, because django channels creates a lot of different event-loops with async_to_sync. This needs more testing